lib/core/board.rb

```diff
@@ -0,0 +1,63 @@
+require "core/tile"
+
+class Board
+  attr_reader :board_size
+  attr_reader :tiles
+
+  def initialize(board_size = 3) 
+    @board_size = board_size
+    @tiles = create_tiles
+  end
+
+  def available_moves
+    available_tiles.map { |tile| tile.number }
+  end
+
+  def make_move(tile_number, player_symbol)
+    tile = get_tile(tile_number) 
+    tile.player_symbol = player_symbol unless tile.nil?
+  end
+
+  def is_terminal?
+    is_full? || has_won?(:X) || has_won?(:O)
+  end
+
+  def is_full?
+    tiles.all? { |t| !t.is_empty? }
+  end
+
+  def has_won?(player_symbol)
+    winning_combinations.any? { |c| has_winning_combination?(c, player_symbol) }
```

Comment:
I think it's better not to use single letter variables for anything that's not a simple index. It's obvious what `c` is when you read the method properly, but for someone skimming to get a sense of what's going on it obscures the details and forces you to slow down.